### PR TITLE
Refactoring conditional directives for alternative if statements.

### PR DESCRIPTION
**Pull request by wmarquesr**
A suggestion to compile entire statements and expressions, as suggested by code style guidelines from the Linux Kernel and practitioners.

* [Linux kernel coding style](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/CodingStyle#n892)
* [Living in the #ifdef Hell](https://www.cqse.eu/en/blog/living-in-the-ifdef-hell/)

It might improve code understanding, maintainability and error-proneness.

### CODE REVIEW

**Comment by wmarquesr** [comment]
You can review the PR ?

**Comment by rogerclarkmelbourne** [comment]
Sorry

I've been ultra busy in the run up to Christmas

I'll try to take a look during the weekend

**Comment by wmarquesr** [comment]
@stevstrong This patch suggests an alternative form to use #IFDEFs, enhancing readability and understanding of the code.

**Comment by rogerclarkmelbourne** [comment]
@stevstrong 

I see what you mean about the removal of white space

@wmarquesr 
The problem with removing white space and using Git is that every line you removed white space from is marked as a if its a code change

So "I can't see the wood for the trees",  is there any way you can do this PR again but only change the IFDEF's

BTW. What bug were you fixing by changing the IFDEF's ?
Was there a problem on a specific OS ?

File: GD32F1/libraries/MapleCoOS116/utility/task.c [hunk 1]
```diff
@@ -764,12 +769,7 @@ static inline void _schedule(BOOL yield)
 		RemoveFromTCBRdyList(pRdyTcb);
         pRdyTcb->state = TASK_RUNNING;
     }
-
```
Comment (wmarquesr) [inline GD32F1/libraries/MapleCoOS116/utility/task.c]: Before code.

File: GD32F1/libraries/MapleCoOS116/utility/task.c [hunk 2]
```diff
@@ -731,26 +731,31 @@ StatusType CoSetPriority(OS_TID taskID,U8 priority)
...
  */
-static inline void _schedule(BOOL yield) 
+static inline void _schedule(BOOL yield)
 {
     U8  RunPrio,RdyPrio;
     P_OSTCB pRdyTcb,pCurTcb;
-   
-	
-    pCurTcb = TCBRunning;    
+
```
Comment (wmarquesr) [inline GD32F1/libraries/MapleCoOS116/utility/task.c]: After code.

File: GD32F1/libraries/MapleCoOS116/utility/task.c [hunk 3]
```diff
@@ -731,26 +731,31 @@ StatusType CoSetPriority(OS_TID taskID,U8 priority)
...
+#if CFG_ROBIN_EN > 0  /* Is time for robinning                  */
+  #define PRIO_STATUS ((RunPrio == RdyPrio) && (yield || (OSCheckTime == OSTickCnt)))
+#else /* simple yield */
+  #define PRIO_STATUS ((RunPrio == RdyPrio) && yield)
+#endif
+
```
Comment (wmarquesr) [inline GD32F1/libraries/MapleCoOS116/utility/task.c]: After code.

File: GD32F1/libraries/MapleCoOS116/utility/task.c [hunk 4]
```diff
@@ -764,12 +769,7 @@ static inline void _schedule(BOOL yield)
 		RemoveFromTCBRdyList(pRdyTcb);
         pRdyTcb->state = TASK_RUNNING;
     }
-    
-#if CFG_ROBIN_EN >0                 /* Is time for robinning                  */                            
-	else if ((RunPrio == RdyPrio) && (yield || (OSCheckTime == OSTickCnt)))  
-#else /* simple yield */  
-    else if ((RunPrio == RdyPrio) && yield)  
-#endif
```
Comment (wmarquesr) [inline GD32F1/libraries/MapleCoOS116/utility/task.c]: Before code.

File: STM32F1/libraries/MapleCoOS116/utility/task.c [hunk 5]
```diff
@@ -731,26 +731,31 @@ StatusType CoSetPriority(OS_TID taskID,U8 priority)
...
-   
-	
-    pCurTcb = TCBRunning;    
+
+#if CFG_ROBIN_EN > 0  /* Is time for robinning                  */
+  #define PRIO_STATUS ((RunPrio == RdyPrio) && (yield || (OSCheckTime == OSTickCnt)))
+#else /* simple yield */
+  #define PRIO_STATUS ((RunPrio == RdyPrio) && yield)
+#endif
+
```
Comment (wmarquesr) [inline STM32F1/libraries/MapleCoOS116/utility/task.c]: After code.

File: STM32F1/libraries/MapleCoOS116/utility/task.c [hunk 6]
```diff
@@ -764,12 +769,7 @@ static inline void _schedule(BOOL yield)
 		RemoveFromTCBRdyList(pRdyTcb);
         pRdyTcb->state = TASK_RUNNING;
     }
-    
-#if CFG_ROBIN_EN >0                 /* Is time for robinning                  */                            
-	else if ((RunPrio == RdyPrio) && (yield || (OSCheckTime == OSTickCnt)))  
-#else /* simple yield */  
-    else if ((RunPrio == RdyPrio) && yield)  
-#endif
```
Comment (wmarquesr) [inline STM32F1/libraries/MapleCoOS116/utility/task.c]: Before code.

**Comment by wmarquesr** [comment]
Better now ?

**Comment by rogerclarkmelbourne** [comment]
When I click on https://github.com/rogerclarkmelbourne/Arduino_STM32/pull/153/files it still shows the stuff to do with white space

Re: Your ifdef changes, you seem to have changed 

>0

for 

> 0

Is this general scope of the change

Can you explain why 

>0 doesnt work ?

**Comment by rogerclarkmelbourne** [comment]
I'm closing this as I've not had a reply for 2 months

**Pull request closed without merging**